collector/collector.go

```diff
@@ -119,7 +123,14 @@ func (c *Collector) Collect() (*Result, error) {
 
 		res, err := httpClient.Do(req)
 		if err != nil {
+			// Authorization error signals token is invalid
 			return nil, err
+		} else {
+			// non-2xx status code don't cause errors so we branch off to check
+			// for them
+			if res.StatusCode == 401 {
+				return nil, fmt.Errorf("http 401 response received %w", ErrUnauthorized)
+			}
```

Comment:
Because the if condition has a `return` in it, the `else` is redundant, and it's a bit of an unwritten part of our style guide that we don't do this.

A non-stylistic issue is that if `res.StatusCode == 401`, the body of `res` won't be called when the function method, which may hold the connection open and leak memory.

It's not that bad because we know the main method will exit, but I think we need to preserve conventions like the expectation that the response body will be closed when the method returns.

Reply:
You can fix both by moving the 
```
			if res.StatusCode == 401 {
				return nil, fmt.Errorf("http 401 response received %w", ErrUnauthorized)
			}
```
part after the `defer` and not using an `else`.